kcidb/oo/__init__.py

```diff
@@ -406,6 +406,88 @@ def waived_status_tests(self):
+                if platform in sorted_tests:
+                    existing_tests = sorted_tests[platform].get(status)
+                    if existing_tests:
+                        sorted_tests[platform].update({
+                            status: existing_tests.append(test)})
+                    else:
+                        sorted_tests[platform].update({status: list(test)})
+                else:
+                    sorted_tests[platform] = {status: list(test)}
+        return sorted_tests
+
+    def get_fluster_stats(self, sorted_tests_by_platform):
+        """
+        Get number of passed and failed tests for all enabled fluster tests
+
+        Argument: (sorted_tests_by_platform) A dictionary with key as a
+        platform and value as a dictionary with status as a key and test
+        list as a value in the below format:
+        {<platform>: {<status1>: [test1], <status2>: [test2, test3]}
+
+        Returns: (final_stats) A dictionary with key as a platform and value
+        as a list consists of fluster tests stats in the below format:
+        {<platform>: [[<fluster-test1>, <passed-count>,
+        <failed-count>, <total_count>], [<fluster-test2>, <passed-count>,
+        <failed-count>, <total_count>]]}
+        """
+        fluster_tests_mapping = {
+            "fluster.v4l2.gstreamer_av1": "v4l2-decoder-conformance-av1",
+            "fluster.v4l2.gstreamer_av1_chromium":
+                "v4l2-decoder-conformance-av1-chromium-10bit",
+            "fluster.v4l2.gstreamer_h264": "v4l2-decoder-conformance-h264",
+            "fluster.v4l2.gstreamer_h264_frext":
+                "v4l2-decoder-conformance-h264-frext",
+            "fluster.v4l2.gstreamer_h265": "v4l2-decoder-conformance-h265",
+            "fluster.v4l2.gstreamer_vp8": "v4l2-decoder-conformance-vp8",
+            "fluster.v4l2.gstreamer_vp9": "v4l2-decoder-conformance-vp9"
+        }
+        final_stats = {}
+        for platform, data in sorted_tests_by_platform.items():
+            stats = []
+            total = [test for tests in data.values() for test in tests]
+            for key, value in fluster_tests_mapping.items():
+                passed_tests = [t.path for t in data.get("PASS", [])
+                                if key in t.path and t.path != key and
+                                t.path != f"{key}.validate-fluster-results"]
+                failed_tests = [t.path for t in data.get("FAIL", [])
+                                if key in t.path and t.path != key and
+                                t.path != f"{key}.validate-fluster-results"]
+                total_tests = [t.path for t in total
+                               if key in t.path and t.path != key and
+                               t.path != f"{key}.validate-fluster-results"]
+                stats.append([value, len(passed_tests), len(failed_tests),
+                              len(total_tests)])
+            final_stats[platform] = stats
+        return final_stats
+
```

Comment:
All these three functions have no business being in the `Node` class, they never access `self`. Furthermore, they are too specific to reside in such a fundamental class. Finally, this is all possible to do in Jinja2 directly and with less complexity.

I added a commit replacing these (roughly) with Jinja2 code. Some counts don't match, but I don't have any more time for teasing that out, and would leave that to you, as well as making the formatting nicer.